---
title: "Introduction to the Workshop"
pagetitle: "Workshop Intro"
highlight-style: github
freeze: true
author:
  - Jelmer Poelstra
  - Mike Sovic
---

```{r knitr_options, echo=FALSE}
knitr::opts_chunk$set(eval = TRUE, echo = FALSE,
                      warning = FALSE, message = FALSE,
                      class.output = "bash-out")
```

-----

<br>

## What you will learn

The focus of this workshop is building some **general (foundational) skills**
for analyzing genomics data &mdash;
specifically, for doing so with command-line programs at the Ohio Supercomputer
Center (OSC).

Command-line programs are preferred for many of the steps to analyze genomic
sequencing data.
Because such datasets tend to contain _a lot_ of data,
it is also preferable to run your analyses not on a laptop or desktop,
but at a supercomputer like OSC.

These realities mean that in the field of genomics,
you need the following set of skills
that you may not have been taught previously:

1. Having a basic understanding of a **supercomputer**

And being able to:

2. Use the **Unix shell** (work in a terminal)
3. Write small **shell scripts** that **run command-line programs**
4. **Submit scripts** to a supercomputer's "queue" &mdash;
   and monitor & manage the resulting batch jobs
5. Activate and probably install **software** in a Linux environment where you
   don't have admin rights
  
We will teach the basics of these skills during this workshop!

:::{.callout-caution}
## What you won't learn

It may be useful to point out that we will **not** teach you much,
if anything, about:

- Details of genomic data file types &mdash; except, briefly, FASTQ
- Details of specific (genomic) analyses
- Making biological inferences from your data
:::

<br>

## Practicalities

We have a slightly complicated set up with participants in-person in Wooster
~~with an instructor~~ (Jelmer now via Zoom),
in-person in Columbus with an instructor,
and directly via Zoom.
Some notes:

- The workshop consists of a series of **9 modules**:
  see the [schedule](../index.qmd).
  We intend to respect the start and end times for every day,
  but individual modules may take shorter or longer than indicated below.
  
- We have one 10-15 minute break between two modules every day,
  and another impromptu bathroom break in a longer module.
  
- The instructors will be available for **additional questions**
  from about 15 minutes before we start,
  and for about 30 minutes after we end each day.

- **This website has all the material** that we will go through,
  with one page for each of module.
  See the links in the [schedule](../index.qmd) as well as in the top bar menus
  to access it.

- In-person participants don't need to connect to the Zoom call,
  since Zoom will be broadcast on the large screen
  (but you _can_ of course connect
  if you can better see the instructor's screen that way).

- Because we're not all on Zoom,
  we'll try to avoid the Zoom chat and instead use
  **[this Google Doc](https://docs.google.com/document/d/1M_BSRtnG3QYjntpgxbMENr_ZJGrA35B1e7E0Dn0hK5Q/edit?usp=sharing)**
  to share links, inpromptu code that is not on the website,
  and non-urgent questions.
  
- Whenever you have a **question**,
  please feel free to interrupt and speak up,
  both in-person and on Zoom.
  Because we will mute the in-person rooms on the Zoom call by default,
  signal to Mike (Columbus) or Menuka (Wooster) when you have a question,
  who will then unmute the room.
  
  If your question is not urgent and you don't want to interrupt the flow,
  put it in the Google Doc or ask about it during a break.

<br>

## Sign-up form responses

```{r}
source(here::here("scripts/participants-datatypes.R"))
```

```{r}
ggplot(dt_df) +
  aes(y = data_type, x = count) +
  geom_col(fill = "darkred", color = "grey10") +
  scale_x_continuous(expand = expansion(mult = c(0, 0.03))) +
  labs(y = NULL,
       title = "Data types among participants") +
  theme(panel.grid.major.y = element_blank())
```

<br>

```{r}
## Unix shell
p_unix <- signup %>%
  filter(!is.na(experience_unix)) %>%
  ggplot(aes(y = experience_unix, fill = experience_unix)) +
  geom_bar(color = "grey20") +
  scale_fill_brewer(palette = "Greens", direction = -1) +
  guides(fill = "none") +
  labs(x = NULL, y = NULL, title = "The Unix shell") +
  theme(panel.grid.major.y = element_blank())

## OSC
p_osc <- signup %>%
  filter(!is.na(experience_OSC)) %>%
  ggplot(aes(y = experience_OSC, fill = experience_OSC)) +
  geom_bar(color = "grey20") +
  scale_fill_brewer(palette = "Greens", direction = -1) +
  guides(fill = "none") +
  labs(x = NULL, y = NULL, title = "OSC") +
  theme(panel.grid.major.y = element_blank(),
        axis.text.y = element_blank())

## Coding
p_coding <- signup %>%
  filter(!is.na(experience_coding)) %>%
  ggplot(aes(y = experience_coding, fill = experience_coding)) +
  geom_bar(color = "grey20") +
  scale_fill_brewer(palette = "Greens", direction = -1) +
  guides(fill = "none") +
  labs(y = NULL, title = "Coding") +
  theme(panel.grid.major.y = element_blank())

## Genomics
p_genomics <- signup %>%
  filter(!is.na(experience_genomics)) %>%
  ggplot(aes(y = experience_genomics, fill = experience_genomics)) +
  geom_bar(color = "grey20") +
  scale_fill_brewer(palette = "Greens", direction = -1) +
  guides(fill = "none") +
  labs(y = NULL, title = "Genomics") +
  theme(panel.grid.major.y = element_blank(),
        axis.text.y = element_blank())

(p_unix + p_osc) / (p_coding + p_genomics) +
   plot_annotation(title = "Participants' experience level with:")
```

<br>

## Personal introductions

### Instructors

- **Jelmer Poelstra**, Molecular and Cellular Imaging Center (MCIC), Wooster
- **Mike Sovic**, Center for Applied Plant Sciences (CAPS), Wooster

A special mention goes out to _Menuka Bhandari_ who is managing the Selby
in-person room. Menuka is also knowledgeable about the workshop material
and may be able to help you if you need/prefer someone in the room itself
for a certain question.

### Participants

Please very briefly introduce yourself &mdash;
include your position, department, and why you wanted to go to this workshop.
