---
title: "Command line basics <br> for genomic analysis at OSC"
subtitle: "A workshop by the MCIC on August 17 - 19, 2022."
---

## Practicalities

This workshop will be taught **in-person at the Wooster OSU Campus**
by Jelmer Poelstra from the Molecular and Cellular Imaging Center (MCIC).

The workshop will be highly hands-on and take place across three afternoons:  
**Wed, Aug 17 - Fri, Aug 19 from 1 pm to 4:30 pm**.

Otherwise:

- Anyone affiliated with The Ohio State University or Wooster USDA is welcome to sign up
- Attendance is free
- No prior experience with coding or genomic data is required
- You will only need to bring a laptop and don't need to install anything
  prior to or during the workshop

See below for information about the contents of the workshop and to sign up.  
For questions, please [email Jelmer](mailto:poelstra.1@osu.edu).

## Contents of the workshop

**The workshop is geared towards people who would like to get started**
**with analyzing genomic datasets.**

The focus of the workshop is not on any specific data or analysis type but
on teaching the general skills you need to be able to analyze almost _any_
genomics dataset.
These boil down to the ability to write small scripts that run
command-line programs, and submit these scripts to a remote computer --
in our case, the Ohio Supercomputer Center (OSC).

These skills can in fact be applied to working with any kind of (large) dataset.
However, on our campus, by far the most common use case for command-line computing
at OSC is genomic data -- such as from
_RNAseq_, _metabarcoding_, _metagenomic shotgun sequencing_, or _whole-genome sequencing_.

### Modules of the workshop

- Introduction to the Ohio Supercomputer Center (OSC)
- Using the `VS Code` text editor at OSC
- Introduction to the Unix shell (= the terminal / command line)
- Basics of shell scripts
- Data transfer and software at OSC with _modules_ & `Conda`
- Submitting your scripts using the `SLURM` scheduler
- Putting it all together: practical examples of running analysis jobs at OSC

The modules will be a mixture of lectures that include "participatory live-coding"
(i.e. with the instructor slowly demonstrating and participants expected to follow
along for themselves) and exercises.

We will work with example genomics data but if you have any,
you are welcome to bring your own data to put your newly learned skills
in practice on the last afternoon.

### Some more background

When doing any kind of large-scale sequencing experiment,
you will typically receive the raw sequencing data.
Many of the steps to analyze this data, such as those involving quality control,
trimming or adapter removal, and assembly or mapping,
are best performed using command-line programs.
Additionally, _a lot_ of data tends to be involved,
and many analysis steps can be done independently for each sample.
Therefore, it pays off (or may be necessary) to run your analyses not
on a laptop or desktop, but at a supercomputer like OSC.

Being able to run your analysis at OSC involves a number of skills that may seem
overwhelming at first: working in a terminal (the Unix shell),
writing small scripts to run programs with a command-line interface,
submitting scripts to a "queue" and managing the resulting compute jobs,
accessing if not also installing software,
and having some understanding of the supercomputer.

Fortunately, learning the basics of these skills does not take a lot of time
_and_ is sufficient to really be up-and-running with your own genomic data. 
These days, excellent programs are available for almost any genomics analysis.
Therefore, you do not need to be able to code it all up from scratch:
you just need to know how to efficiently run these programs at the command line,
which is what this workshop aims to teach you.

## Sign up!

To sign up, please fill out the form below:

<iframe src="https://docs.google.com/forms/d/e/1FAIpQLScEyRqHKPZXteXE-1nDSz2ahJYK3PwxN47bsIx2XkGSj_wB9A/viewform?embedded=true" width="800" height="1847" frameborder="0" marginheight="0" marginwidth="0">Loadingâ€¦</iframe>

